VirtIO/VirtIORing.c

```diff
@@ -233,7 +236,10 @@ bool virtqueue_enable_cb_delayed(struct virtqueue *vq)
 
     if (!virtqueue_is_interrupt_enabled(vq)) {
         vq->master_vring_avail.flags &= ~VIRTQ_AVAIL_F_NO_INTERRUPT;
-        vq->vring.avail->flags &= ~VIRTQ_AVAIL_F_NO_INTERRUPT;
+        if (!vq->event_suppression_enabled)
```

Comment:
if the VIRTIO_F_EVENT_IDX feature bit is negotiated:
• The driver MUST set flags to 0.

I kept the shadow copy for reference in case we need to debug this. However thinking about it again maybe I should update both of them.

Reply:
The idea with keeping the shadow copy for virtqueue_is_interrupt_enabled() is fine. I just failed to understand the intent of the code.

Maybe something like:

```
static inline set_avail_flags(struct virtqueue *vq, u16 flags)
{
    if (!vq->event_suppression_enabled) {
        vq->vring.avail->flags = flags;
    }
}
```

and call it from all three places? Thanks!